src/MongoDB.Analyzer/Core/Linq/LinqAnalyzer.cs

```diff
@@ -113,7 +114,9 @@ private static AnalysisStats ReportMqlOrInvalidExpressions(MongoAnalysisContext
                 if (isDriverOrLinqException || settings.OutputInternalExceptions)
                 {
                     var diagnosticDescriptor = LinqDiagnosticsRules.DiagnosticRuleNotSupportedLinqExpression;
-                    var decoratedMessage = DecorateMessage(mqlResult.Exception.InnerException?.Message ?? "Unsupported LINQ expression", driverVersion, settings);
+                    var typesMapper = new TypesMapper(MqlGeneratorSyntaxElements.Linq.MqlGeneratorNamespace, context.TypesProcessor);
```

Comment:
Reuse the instance. And skip remapping if there is no exception message.

Reply:
Made it a shared instance among Builders/LINQ/POCO processing